04-onion-routing.md

```diff
@@ -1057,9 +1057,16 @@ The _erring node_:
 The _origin node_:
   - once the return message has been decrypted:
     - SHOULD store a copy of the message.
-    - SHOULD continue decrypting, until the loop has been repeated 20 times.
+    - SHOULD continue decrypting, until the loop has been repeated 27 times
+    (maximum route length of tlv payload type).
     - SHOULD use constant `ammag` and `um` keys to obfuscate the route length.
 
+### Rationale
+
+The requirements for the _origin node_ should help hide the payment sender. By
+continuing decrypting 27 times (dummy decryption cycles after the error is found)
+no information can be obtained via timing analysis.
```

Comment:
Doesn't the effort for Attributable Errors directly contradict this? This is definitely the weakest part of my spec knowledge, but afaict, the choice to continue decrypting beyond the required amount doesn't actually materially affect privacy right?

I understand this is a rationale expansion and not a fundamental change to the spec, but I'm just trying to make sure I actually understand the change to be able to evaluate the correctness of the added rationale section.

Reply:
Good question. Yes as stated in the issue it's just a should and tcp-network latency is probably way more dominant than decrypting some more hops. But what I understand is that we don't want to reveal the `erring node` its position in the route, which could on the one hand reveil the `sender` but on the other hand  also the destination. See also this comment: https://github.com/lightningnetwork/lightning-onion/blob/master/crypto.go#L351-L354

Maybe we need to be even more specific in the rationale how this timing analysis can harm the sender/receiver.

Reply:
hmm yes but we don't want the erring_node to learn its position in the route when it's deliberately failing the route. Which could be done if the erring_node fails some payments and the sender retries them with a smaller amount maybe ?

Reply:
Let's maybe amend the rationale section as follows:

```
...decrypting 27 times (...) the erroring node cannot learn its relative in position in the route by performing a timing analysis if the sender were to retry the same route multiple times
```

because with the current proposed wording I wasn't able to tell who would be doing the timing analysis to learn what.